inst/pages/exercises.qmd

```diff
@@ -1674,71 +1692,126 @@ p <- Heatmap(
+# Perform k-means clustering (k = 3) using the base R kmeans function
+set.seed(123)
+kmeans_result <- kmeans(mds_data, centers = 3)
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
-```
+# Create KmeansParam object
+kmeans_param <- KmeansParam(centers = 3)
 
-Where:
-- `reducedDim` is a function that extracts the reduced dimensionality
-representation of your data (e.g., from a PCA or PCoA object).
-- `plotReducedDim` is a function that can plot these reduced dimensions and
-has a 'colour_by argument' that accepts a vector of cluster assignments.
+# Visualize MDS with clusters
+plotReducedDim(tse, dimred = "MDS") +
+  labs(title = "MDS with k-means Clustering") +
+  theme_minimal()
 
-#### 2. Heatmap with Annotations:
+3. Hierarchial Clustering
+- Hierarchical clustering of samples based on abundance data.
+- Visualizes the clustered data using a heatmap, with cluster information added to metadata.
 
-- Annotate the heatmap with additional metadata (e.g., subject groups or
-sample types).
+```{r}
+#| label: extra-hierarchialclustering-heatmap-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-```{r, eval = FALSE}
 # Load necessary libraries
 library(SummarizedExperiment)
-library(ComplexHeatmap)
+library(sechm)
 
-# Example data creation
-# Assuming enterotype is a SummarizedExperiment object with sample metadata
-# and assay_data is a matrix containing assay data
+# Generate sample abundance data
+set.seed(123)
+sample_data <- matrix(rnorm(100), nrow = 10)
+rownames(sample_data) <- paste0("Gene", 1:10)
+colnames(sample_data) <- paste0("Sample", 1:10)
+
+# Perform hierarchical clustering
+hclust_result <- hclust(dist(t(sample_data)), method = "complete")
+
+# Assign cluster information to metadata
+clusters <- cutree(hclust_result, k = 3)
+metadata <- data.frame(Cluster = as.factor(clusters))
+rownames(metadata) <- colnames(sample_data)
+
+# Create a SummarizedExperiment object
+se <- SummarizedExperiment(assays = list(counts = sample_data),
+                           colData = metadata)
 
-# Create example SummarizedExperiment object
-example_data <- matrix(runif(200), nrow = 20, ncol = 10) # Example assay data
-col_data <- DataFrame(
-    Sample = paste0("Sample", 1:10),
-    Group = rep(c("A", "B"), each = 5)) # Example sample metadata
-enterotype <- SummarizedExperiment(
-    assays = list(counts = example_data), colData = col_data)
+# Visualization using sechm heatmap
+sechm(se,
+      features = rownames(sample_data),
+      cluster_rows = TRUE,
+      cluster_cols = TRUE,
+      )
+
+4. Heatmap with Custom Color Palette:
+Customize the heatmap with a specific color palette to enhance visualization using ComplexHeatmap.
+
+```{r}
+#| label: extra-customcolorheatmap-ex-4
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+# Load necessary libraries
+library(ComplexHeatmap)
```

Comment:
Should we use sechm or ComplexHeatmap?

Reply:
ComplexHeatmap gives nice cutom color palette visuals, but sechm could be used too.

Reply:
Ok I see. I think sechm is ok.

Why this has "eval: false"? Could we have it evaluated (true)?